src/components/Panel/Panel.tsx

```diff
@@ -0,0 +1,101 @@
+/** @jsx h */
+
+import { h } from 'preact';
+import { useState, useEffect, useRef } from 'preact/hooks';
+import cx from 'classnames';
+import Template from '../Template/Template';
+import { PanelCSSClasses, PanelTemplates } from '../../widgets/panel/panel';
+import { RenderOptions } from '../../types';
+
+type PanelComponentCSSClasses = Omit<
+  {
+    [TKey in keyof PanelCSSClasses]: string;
+  },
+  // `collapseIcon` is only used in the default templates of the widget
+  'collapseIcon'
+>;
+
+interface PanelProps {
+  hidden: boolean;
+  collapsible: boolean;
+  isCollapsed: boolean;
+  data: RenderOptions;
+  cssClasses: PanelComponentCSSClasses;
+  templates: PanelTemplates;
+  bodyElement: HTMLElement;
+}
+
+function Panel(props: PanelProps) {
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(props.isCollapsed);
+  const [isControlled, setIsControlled] = useState<boolean>(false);
+  const bodyRef = useRef<HTMLElement | null>(null);
+
+  useEffect(() => {
+    if (!bodyRef.current) {
+      return;
+    }
+
+    bodyRef.current.appendChild(props.bodyElement);
+  }, [bodyRef, props.bodyElement]);
```

Comment:
If I understand correctly,
whenever either `bodyRef` or `props.bodyElement` changes, this effect runs.
`bodyRef` itself is not going to change, but what if `props.bodyElement` changes?
Will it append it again? or it won't happen by design?

Reply:
It won't be design but you're right, we should probably clean the effect:

```ts
return () => {
  bodyRef.current.removeChild(props.bodyElement);
};
```

Reply:
Addressed in e983babb853b56e5ff47f7622d13afa2287392a8.